src/modules/users/user.service.ts

```diff
@@ -0,0 +1,110 @@
+import { Not } from 'typeorm';
+import jwt from 'jsonwebtoken';
+import * as bcrypt from 'bcrypt';
+import { plainToInstance } from 'class-transformer';
+
+import { dataSource } from '../../db';
+import { Gender, User } from './user.schema';
+import { CreateUserRequest } from './dto/CreateRequest';
+import { validationEntity } from '../../middlewares/validate';
+import { UpdateUserRequest } from './dto/UpdateRequest';
+
+export class UserService {
+  private userRepository = dataSource.getRepository(User);
+
+  async createUser(data: CreateUserRequest) {
+    const encrypthPassword = await bcrypt.genSalt();
+    const hashedPassword = await bcrypt.hash(data.password, encrypthPassword);
+
+    const user = this.userRepository.create({
+      name: data.name,
+      email: data.email,
+      phone: data.phone,
+      gender: data.gender,
+      password: hashedPassword,
+    });
+    const entity = plainToInstance(User, user);
+    const validationResult = await validationEntity(User, entity);
+
+    if (validationResult.sucsses === false) {
+      throw new Error(
+        `User with name: ${data.name}, Failed to create: ${validationResult.errors}`
+      );
+    }
+    return await this.userRepository.save(user);
+  }
+
+  async signIn(authCredentials: { phone: string; password: string }) {
+    const { phone, password } = authCredentials;
+    const user = await this.userRepository.findOne({
+      where: { phone, active: true },
+    });
+
+    if (user && (await bcrypt.compare(password, user.password))) {
+      const payload = {
+        id: user.id,
+        role: user.role,
+        departmentId: user.department,
+      };
+      const token = jwt.sign(payload, process.env.JWT_SECRET ?? '', {
+        expiresIn: '30d',
```

Comment:
please move it to .env file
```suggestion
        expiresIn: process.env.TOKEN_EXPIRESTION
```